src/lib/components/SpaceSelect.svelte

```diff
@@ -5,23 +5,22 @@
   import { bg, ColorSpace, fg, format } from '$lib/stores';
   import { getSpaceFromFormatId } from '$lib/utils';
 
-  let spaces: ColorSpace[] = [];
-
-  $: spaces = FORMATS.map((s) => {
+  let spaces: ColorSpace[] = FORMATS.map((s) => {
     if (s === 'hex') return { id: 'hex', name: 'Hex' } as ColorSpace;
     return ColorSpace.get(s);
   });
-  $: targetSpace = getSpaceFromFormatId($format);
 
   // Update color formats when space selection changes
-  $: {
+  // Use $effect.pre to change colors before the $format update is applied.
```

Comment:
Otherwise, `$format` updates globally, which triggers the sliders to update, but because the colors aren't in the new format, the channels don't align and it throws an error. We needed to update the colors before the sliders update.